packer-plugin-sdk/packer/ui_mock.go

```diff
@@ -1,47 +1,18 @@
 package packer
 
 import (
-	"errors"
+	"bytes"
 	"io"
-	"log"
-	"strings"
+	"io/ioutil"
+	"testing"
 )
 
-// TestUi is a UI that only outputs unformatted output to the given writer.
-type TestUi struct {
-	Writer io.Writer
-}
-
-var _ Ui = new(TestUi)
-
-func (u *TestUi) Ask(query string) (string, error) {
-	return "", errors.New("Test UI can't ask")
-}
-
-func (u *TestUi) Say(message string) {
-	log.Printf(message)
-}
-
-func (u *TestUi) Message(message string) {
-	log.Printf(message)
-}
-
-func (u *TestUi) Error(message string) {
-	log.Printf(message)
-}
-
-func (u *TestUi) Machine(message string, args ...string) {
-	log.Printf("%s, %s", message, strings.Join(args, ", "))
-}
-
-func (u *TestUi) TrackProgress(src string, currentSize, totalSize int64, stream io.ReadCloser) (body io.ReadCloser) {
-	return &readCloser{
-		read: func(p []byte) (int, error) {
-			return stream.Read(p)
-		},
-		close: func() error {
-			return stream.Close()
-		},
+func TestUi(t *testing.T) Ui {
```

Comment:
This one could also be moved to its own testing package, so that its clearer that it's meant to be used for tests only in godocs.

Reply:
That's a good point.